src/end-to-end/README.md

```diff
@@ -0,0 +1,33 @@
+# End-to-end tests
+
+## Local runs
+
+See CONTRIBUTING.md for instructions. **TLDR:** the `npm` command outlined in `package.json > scripts` runs the bash file located in this directory.
+
+## CI runs
+
+The CI runs for the end-to-end tests are defined in the `.github` directory.
+
+The `Action` in Github is called `End to end testing`.
+
+- Workflow
+  - The workflow run by CI is called `workflows/e2d-testing.yml`
+  - It requires a Pinecone API key and a Vercel token.
+    - The Pinecone API key it uses the associated with the `ops@` account and is stored in a Github secret
+    - The Vercel token is from the Pinecone Enterprise account and is stored in a Github secret
+- Action
+  - The action run by the workflow is called `actions/e2e-testing/edge/action.yml`
+  - This action packages the `ts-client` code on the current branch into the external app's repo
+  - It then installs the Vercel CLI and sends a POST request to the test app's `/api/createSeedQuery` endpoint,
+    which is up and running on Vercel
+  - If the response from the endpoint does not include a match (a match to a query composed by
+    the test app), the action fails
+  - The action then deletes the index that the test app spun up (note: it will attempt to delete the index even if
+    the assertions fail)
```

Comment:
This content doesn't seem that helpful to me. Since a lot of it is a narrative telling of the workflow, it seems likely to quickly go stale as CI configurations evolve. I would probably delete most of this, but add some explanatory comments in the workflow/job/action configs. Keeping that sort of documentation right next to the implementation will give us a better shot at keeping it up to date.

What I think would be useful here are:
- Some conceptual introduction to what these tests are and why we need them, how they differ from other integration tests, etc.
- Specific instructions for how to run locally. What environment variables are needed and how do I set them? 
- Links to the workflow source and github UI for the workflow.